### DISCUSSION TITLE
413 Payload Too Large

### DISCUSSION THREAD

**Discussion by aimahusnain** [vercel/community#7040] [Help] [answered]
### Summary

Hello, I'm running into the error message: 413 Payload Too Large.

My site is dedicated to background removal and uses FastAPI with Python, Next.js, and TypeScript. For the background removal process, I rely on rembg.

Everything works fine locally, but after deploying to Vercel, it doesn't run properly when the API is called. When I check the network activity, I see the Payload Too Large error.

Here is my git repo: https://github.com/aimahusnain/Background-Removing
And here is my Vercel deployed website that is not running:  https://removebg-delta.vercel.app/

### Steps to Reproduce

![2024-05-30_22-55](https://github.com/vercel/community/assets/103668742/7ac0d097-2c07-49cc-8c8c-92bf8e67ae42)

**Comment by versecafe** [comment, marked as answer]
@aimahusnain [here's a list of solutions](https://vercel.com/guides/how-to-bypass-vercel-body-size-limit-serverless-functions), this is a common issue with file uploads through Vercel since images tend to break the 4.5MB limit, if none of these work for you I can try and help you find an alternative solution.

**Comment by aimahusnain** [comment]
I'm new to FastAPI and have been watching YouTube tutorials and exploring on GitHub. However, I haven't found many projects to study. My learning strategy is to review others' code to understand their approach.

I'm encountering an issue when trying to upload a file smaller than 4MB. On the deployed website, I get a **405 Method Not Allowed** error, but this doesn't happen locally.

Does anyone know how to solve this? Here's the link to my API GitHub repository: [GitHub Repo](https://github.com/aimahusnain/Background-Removing/blob/main/api/index.py).


![2024-05-31_16-33](https://github.com/vercel/community/assets/103668742/772ecbc3-5cdd-4274-ae07-51229b1f21dd)
